scout/adapter/mongo/case.py

```diff
@@ -555,6 +557,61 @@ def delete_case(self, case_id=None, institute_id=None, display_name=None):
         result = self.case_collection.delete_one(query)
         return result
 
+    def check_existing_data(self, case_obj, existing_case, institute_obj, update, keep_actions):
+        """Make sure data from case to be loaded/reuploaded conforms to case data already saved in database.
+           Return eventual evaluated variants to be propagated to the updated case if keep_actions is True
+
+        Args:
+            case_obj(dict): case dictionary to be loaded/reuploaded
+            existing_case(dict): a case with same _id or same display_name and institute_id as case_obj
+            institute_obj(dict): institute dictionary
+            update(bool): If existing case should be updated
+            keep_actions(bool): If old evaluated variants should be kept when case is updated
+
+        Returns:
+            previous_evaluated_variants(list): list of variants evaluated in previous case
+                or None if case is not already present in the database.
+        """
+
+        if existing_case is None:
+            return
+
+        # Avoid saving a new case in database with a display_name that already exists for another case from the same customer
+        if existing_case["_id"] != case_obj["_id"]:
+            raise IntegrityError(
+                "A case with display name %s already exists in database for this customer"
```

Comment:
But you are checking _id, not display name here?

Reply:
Check later, at line 590

Reply:
Ah, or rather line 655! How about printing the _ids as well: that might be handy. That is just QoL though.😊  
I would also rather drop the 579 comment, or explicitly state that it is assumed that the only other way the _ids should differ is If the display name already existed.

Reply:
Thanks! 😊